### PR TITLE
Resource Ctls share commands through embedding

### PR DESCRIPTION
**Pull request by mphammer**
Synopsysctl's edit was always invoking the kubectl editor even when flags were set. The correct functionality is to invoke the kubectl vim editor if no flags are set and just update the resource if flags are set.

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **mphammer**

If they are not already assigned, you can assign the PR to them by writing `/assign @mphammer` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/alert/ctl_alert.go [hunk 1]
```diff
@@ -147,6 +147,33 @@ func (ctl *Ctl) AddSpecFlags(cmd *cobra.Command, master bool) {
 	cmd.Flags().StringVar(&ctl.DesiredState, "alert-desired-state", ctl.DesiredState, "State of the Alert")
 }
 
+var nSpecFlagCmd cobra.Command // command for getting flags for this spec
```
Comment (rrati) [inline pkg/alert/ctl_alert.go]: Why declare as a global variable?  This seems to be used temporarily vs needing to keep the value.  Just have temp variables where this is used assigned with :=.

File: pkg/blackduck/ctl_blackduck.go [hunk 2]
```diff
@@ -225,6 +225,33 @@ func (ctl *Ctl) AddSpecFlags(cmd *cobra.Command, master bool) {
 	cmd.Flags().StringVar(&ctl.LicenseKey, "license-key", ctl.LicenseKey, "License Key of Black Duck")
 }
 
+var nSpecFlagCmd cobra.Command // command for getting flags for this spec
+var specFlags *pflag.FlagSet   // set of flags for this spec
+var numSpecFlagsChanged int    // number of flags that were set for the spec
+
+// NSpecFlag returns the number of spec flags that were set
+func (ctl *Ctl) NSpecFlag(flagset *pflag.FlagSet) int {
```
Comment (rrati) [inline pkg/blackduck/ctl_blackduck.go]: This is the same function as other commands.  Suggest moving into a base class that can be included in these separate command classes.  No reason to duplicate code.

File: pkg/opssight/ctl_opssight.go [hunk 3]
```diff
@@ -279,6 +279,33 @@ func (ctl *Ctl) AddSpecFlags(cmd *cobra.Command, master bool) {
 	cmd.Flags().IntVar(&ctl.BlackduckMaxCount, "blackduck-max-count", ctl.BlackduckMaxCount, "Maximum number of Black Ducks that can be created")
 }
 
+var nSpecFlagCmd cobra.Command // command for getting flags for this spec
+var specFlags *pflag.FlagSet   // set of flags for this spec
+var numSpecFlagsChanged int    // number of flags that were set for the spec
+
+// NSpecFlag returns the number of spec flags that were set
+func (ctl *Ctl) NSpecFlag(flagset *pflag.FlagSet) int {
```
Comment (rrati) [inline pkg/opssight/ctl_opssight.go]: Might as well make the function name descriptive, like NumNewFlags or something.

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/267#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@mphammer: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request closed without merging**